src/ucp/api/ucp.h

```diff
@@ -1187,21 +1200,33 @@ void ucp_worker_wait_mem(ucp_worker_h worker, void *address);
  * or newly available resources. It can also be triggered
  * by calling @ref ucp_worker_signal .
  *
+ * The file descriptor is guaranteed to become signaled only if new communication
+ * events occur on the @a worker. Therefore one must drain all existing events
+ * before waiting on the file descriptor. This can be achieved by by calling
+ * @ref ucp_worker_progress repeatedly until it returns 0.
+ *
  * @code {.c}
  * void application_initialization() {
  *     ...
  *     status = ucp_worker_get_efd(worker, &fd);
  *     ...
  * }
+ *
  * void process_comminucation() {
  *     for (;;) {
- *         ucp_worker_progress(worker);
- *         check_for_events();              // receive() operations
- *         status = ucp_worker_arm(worker); // arm the worker and clean-up fd
+ *         // check for events as long as progress is made
```

Comment:
in this case yes because it's part of the "code example" and we can't put C comment inside a comment